vip-scanner/checks/VIPRestrictedVariablesCheck.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+class VIPRestrictedVariablesCheck extends BaseCheck
+{
+	function check( $files ) {
+		$result = true;
+
+		$checks = array(
+			'((?<![\\\'\"])\$\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*?(?![\\\'\"])|(?<![\\\'\"])\$\{(?:.*)[\}](?![\\\'\"]))' => array( "level" => "Warning", "note" => "Possible variable variables" ),
```

Comment:
There are some other interesting possibilities we can try to account for:

```
${"foo$one"} = 'foo'; 
echo $foo1; // foo 
${'foo' . ONE} = 'bar'; 
echo $foo1; // bar 
${'foo' . one()} = 'baz'; 
echo $foo1; // baz 
```

http://php.net/manual/en/language.variables.variable.php#75669